### PR TITLE
 Compile game.c and game.h as C++

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/Game.cpp [hunk 1]
```diff
@@ -101,38 +103,41 @@ sint32 game_command_playerid = -1;
...
+    {
         return game_command_callback_table[index];
     }
     return 0;
 }
 
 void game_increase_game_speed()
 {
-    gGameSpeed = min(gConfigGeneral.debugging_tools ? 5 : 4, gGameSpeed + 1);
+    gGameSpeed     = Math::Min(gConfigGeneral.debugging_tools ? 5 : 4, gGameSpeed + 1);
```
Comment (janisozaur) [inline src/openrct2/Game.cpp]: this misalignment happens a few times and is not needed.

File: src/openrct2/Game.cpp [hunk 2]
```diff
@@ -1317,58 +1458,69 @@ void save_game_as()
...
 
     file_info fileInfo;
 
     utf8 filter[MAX_PATH];
 
-    utf8 **autosaveFiles = NULL;
+    utf8 ** autosaveFiles = nullptr;
 
-    size_t i=0;
+    size_t i = 0;
```
Comment (janisozaur) [inline src/openrct2/Game.cpp]: this variable should be lowered to the loop that are using it.

**Comment by Gymnasiast** [comment]
@janisozaur Addressed both points.

File: src/openrct2/Game.cpp [hunk 3]
```diff
@@ -279,29 +296,31 @@ void update_palette_effects()
             }
         }
 
-        j = ((uint16)(gPaletteEffectFrame * -960) * 3) >> 16;
+        j       = ((uint16) (gPaletteEffectFrame * -960) * 3) >> 16;
         waterId = SPR_GAME_PALETTE_4;
-        g1 = gfx_get_g1_element(shade + waterId);
-        if (g1 != NULL)
+        g1      = gfx_get_g1_element(shade + waterId);
```
Comment (janisozaur) [inline src/openrct2/Game.cpp]: This doesn't look aligned properly. Your formatting tool commonly mistakes assignments with other uses of `=`, please keep an eye out for that.

File: src/openrct2/Game.cpp [hunk 4]
```diff
@@ -279,29 +296,31 @@ void update_palette_effects()
...
+        if (g1 != nullptr)
         {
             uint8 * vs = &g1->offset[j * 3];
             uint8 * vd = &gGamePalette[243 * 4];
-            sint32 n = 3;
+            sint32      n = 3;
```
Comment (janisozaur) [inline src/openrct2/Game.cpp]: same case here

File: src/openrct2/Game.cpp [hunk 5]
```diff
@@ -312,33 +331,39 @@ void game_update()
...
+    if (gGameSpeed > 1)
+    {
         numUpdates = 1 << (gGameSpeed - 1);
-    } else {
+    }
+    else
+    {
         numUpdates = gTicksSinceLastUpdate / GAME_UPDATE_TIME_MS;
-        numUpdates = clamp(1, numUpdates, GAME_MAX_UPDATES);
+        numUpdates = Math::Clamp(1u, numUpdates, (uint32) GAME_MAX_UPDATES);
```
Comment (janisozaur) [inline src/openrct2/Game.cpp]: Maybe `GAME_MAX_UPDATES` can be made unsigned rather than having to cast it here?

File: src/openrct2/Game.cpp [hunk 6]
```diff
@@ -585,29 +629,35 @@ sint32 game_do_command_p(sint32 command, sint32 *eax, sint32 *ebx, sint32 *ecx,
...
+            if (!(flags & GAME_COMMAND_FLAG_APPLY))
+            {
                 // Decrement nest count
                 gGameCommandNestLevel--;
                 return cost;
             }
 
-            if (network_get_mode() != NETWORK_MODE_NONE && !(flags & GAME_COMMAND_FLAG_NETWORKED) && !(flags & GAME_COMMAND_FLAG_GHOST) && !(flags & GAME_COMMAND_FLAG_5) && gGameCommandNestLevel == 1 /* Send only top-level commands */) {
-                if (command != GAME_COMMAND_LOAD_OR_QUIT) { // Disable these commands over the network
+            if (network_get_mode() != NETWORK_MODE_NONE && !(flags & GAME_COMMAND_FLAG_NETWORKED) && !(flags & GAME_COMMAND_FLAG_GHOST) && !(flags & GAME_COMMAND_FLAG_5) && gGameCommandNestLevel == 1 /* Send only top-level commands */)
```
Comment (janisozaur) [inline src/openrct2/Game.cpp]: Please break this line

File: src/openrct2/Game.cpp [hunk 7]
```diff
@@ -585,29 +629,35 @@ sint32 game_do_command_p(sint32 command, sint32 *eax, sint32 *ebx, sint32 *ecx,
...
+            {
+                if (command != GAME_COMMAND_LOAD_OR_QUIT)
+                { // Disable these commands over the network
```
Comment (janisozaur) [inline src/openrct2/Game.cpp]: Looks a bit odd here, I'd rather expect the comment on its own line

**Comment by janisozaur** [comment]
Some more minor formatting comments, but otherwise LGTM.

File: src/openrct2/Game.cpp (hunk 3 above)
Comment (Gymnasiast) [reply src/openrct2/Game.cpp]: This is aligned properly though. It aligns with `waterId = SPR_GAME_PALETTE_4;`:
```
        j       = ((uint16) (gPaletteEffectFrame * -960) * 3) >> 16;
        waterId = SPR_GAME_PALETTE_4;
        g1      = gfx_get_g1_element(shade + waterId);
        if (g1 != nullptr)
```
I think you're thrown off by the diff.

File: src/openrct2/Game.cpp (hunk 5 above)
Comment (Gymnasiast) [reply src/openrct2/Game.cpp]: Tried it, but doesn't seem to work, sadly.

File: src/openrct2/Game.cpp [hunk 8]
```diff
@@ -1377,21 +1533,22 @@ static void limit_autosave_count(const size_t numberOfFilesToKeep, bool processL
...
     // Calculate how many saves we need to delete.
     numAutosavesToDelete = autosavesCount - numberOfFilesToKeep;
 
-    i=0;
-    while (numAutosavesToDelete > 0) {
+    for (size_t i = 0; numAutosavesToDelete > 0;)
+    {
         platform_file_delete(autosaveFiles[i]);
 
         i++;
```
Comment (duncanspumpkin) [inline src/openrct2/Game.cpp]: may as well move this into the for loop

**Pull request merged**